code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm

```diff
@@ -260,7 +264,69 @@
 						break
 			if (visible_contents && .)
 				update_icon()
+		if("examine")
+			var/name = params["name"]
+			var/obj/item/target
+			for(var/obj/item/O in src)
+				if(O.name == name)
+					target = O
+					break
+
+			if(!target)
+				return
+
+			//Code copypasted from /mob/verb/examinate
+			if(is_blind(usr))
+				to_chat(usr, "<span class='warning'>You can't see inside [src]!</span>")
+				return
+
+			usr.face_atom(src)
+			var/list/result = target.examine(usr)
+			to_chat(usr, result.Join("\n"))
+			SEND_SIGNAL(usr, COMSIG_MOB_EXAMINATE, target)
```

Comment:
doesnt this already get sent when you trigger examine?

Reply:
This is another thing that left me very confused, and is one piece of shitcode that seems necessary here. `examine` doesn't actually examine things, it only builds the strings, it's `examinate` that does the actual printing... But it also has a view check, since it's a verb.

It seems like there are many cases in code where `examine` is called directly, which seems to do *absolutely nothing* as the returned value is discarded. The code here is mostly copypasted from `examinate`, but doesn't let blind people examine items by touching them (as they're in the smartfridge), and ignores view checks.